src/components/Choropleth.vue

```diff
@@ -34,36 +34,39 @@ import { APP_BASE_ROUTE } from "../router/utils";
 import { debounce } from "../utils";
 import { backgroundLayer } from "./utils";
 import { useLoadingSpinner } from "../composables/useLoadingSpinner";
+import { useSelectedMapInfo } from "../composables/useSelectedMapInfo";
 
 const mapLoading = ref(true);
 const router = useRouter();
-const { selectedFeatures, selectedIndicators, selectedIndicator, selectedCountryId, appConfig } =
-    storeToRefs(useAppStore());
+const { mapSettings, appConfig } = storeToRefs(useAppStore());
 const featureProperties = appConfig.value.geoJsonFeatureProperties;
 
 const { tooltipForFeature } = useTooltips();
-const { getFillAndOutlineColour } = useColourScale(selectedIndicators);
+const { getFillAndOutlineColour } = useColourScale();
 
 const featureInSelectedCountry = (feature: Feature, selectedCountry: string) =>
     feature.properties[featureProperties.country] === selectedCountry;
 
-const getFeatureId = (feature: Feature) =>
-    featureInSelectedCountry(feature, selectedCountryId.value)
-        ? feature.properties![featureProperties.idAdm2]
-        : feature.properties![featureProperties.idAdm1];
-
-const getFeatureName = (feature: Feature) =>
-    featureInSelectedCountry(feature, selectedCountryId.value)
-        ? feature.properties![featureProperties.nameAdm2]
-        : feature.properties![featureProperties.nameAdm1];
+const getFeatureProp = (feature: Feature, key: "name" | "id") => {
+    const isAdmin2 = featureInSelectedCountry(feature, mapSettings.value.country) && mapSettings.value.adminLevel === 2;
+    let propertyKey: string;
+    if (key === "id") {
+        propertyKey = isAdmin2 ? featureProperties.idAdm2 : featureProperties.idAdm1;
+    } else {
+        propertyKey = isAdmin2 ? featureProperties.nameAdm2 : featureProperties.nameAdm1;
+    }
+    return feature.properties[propertyKey];
+};
```

Comment:
inspired by change by @EmmaLRussell  https://github.com/mrc-ide/arbomap/pull/12/files#diff-4c9d4b19a4dd499f75587a26f06db5bf0670247d97ed16a2463b93b5cfb2e790R50-R66, basically we werent accounting for countries with no admin2 data which was leading to issues, the `mapSettings` object now accounts for that and the `adminLevel` inside it is set to 1 for countries without admin 2 data so we can just use that now